pkg/collector/corechecks/containers/cri/check_metrics_extension.go

```diff
@@ -0,0 +1,66 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build cri
+// +build cri
+
+package cri
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/containers/generic"
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"github.com/DataDog/datadog-agent/pkg/util/containers/cri"
+	"github.com/DataDog/datadog-agent/pkg/util/containers/v2/metrics/provider"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+
+	criTypes "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+type criCustomMetricsExtension struct {
+	sender            generic.SenderFunc
+	aggSender         aggregator.Sender
+	criGetter         func() (cri.CRIClient, error)
+	criContainerStats map[string]*criTypes.ContainerStats
+}
+
+func (cext *criCustomMetricsExtension) PreProcess(sender generic.SenderFunc, aggSender aggregator.Sender) {
```

Comment:
Should we use in this function or the one below caching (with the 2s defined in the constant) to avoid calling `ListContainerStats` on every call?

Reply:
`PreProcess` is only called once per check run, that's why the `ListContainerStats` is called there instead of `Process`